R/app_server.r

```diff
@@ -69,54 +69,7 @@ app_server <- function(input, output, session) {
       return(shiny::HTML(paste0("<h1>", webpage_title, "</h1>")))
     })
 
-    output$plot_map <- leaflet::renderLeaflet({
-      shiny::req(selection)
-      map_to_plot <- sf::st_transform(healthatlas_data[[selection$atlas()]][["map"]], 32633)
-      map <- helseatlas::make_map(map = map_to_plot, data = selection$data())
-      return(map)
-    })
-
-    output$plot_histogram <- shiny::renderPlot({
-      shiny::req(selection)
-      plot <- helseatlas::plot_variation(
-        input_data = selection$data(),
-        xlab = config$plot$xlab[[input$language]],
-        ylab = "TBA"
-      )
-      return(plot)
-    }
-    , height = 800, width = 600)
-
-    output$make_table <- shiny::renderTable({
-      shiny::req(selection)
-      data_to_tabulate <- selection$data()
-
-      # Return null if data in invalid
-      if (is.null(nrow(data_to_tabulate)) || nrow(data_to_tabulate) == 0) {
-        return(NULL)
-      }
-
-      tabular_data <- data.frame(data_to_tabulate$area_name)
-      colnames(tabular_data) <- c(config$plot$xlab[[input$language]])
-      value_name <- as.character(unique(data_to_tabulate$type))
-      tabular_data[value_name] <- data_to_tabulate$value
-      numerator_name <- as.character(unique(data_to_tabulate$numerator_name))
-      tabular_data[numerator_name] <- data_to_tabulate$numerator
-      denominator_name <- as.character(unique(data_to_tabulate$denominator_name))
-      tabular_data[denominator_name] <- data_to_tabulate$denominator
-      # Sort data
-      tabular_data <- tabular_data[order(tabular_data[, 2], na.last = TRUE, decreasing = TRUE), ]
-      # Format numbers
-      tabular_data[, -1] <- sapply(tabular_data[, -1],
-                                   FUN = function(x) format(x,
-                                                            digits = 2,
-                                                            decimal.mark = config$num$decimal[[input$language]],
-                                                            big.mark = config$num$big[[input$language]]
-                                                            )
-                                   )
-      return(tabular_data)
-    }
-    , align = "lrrr")
+    tab_server("plots")
```

Comment:
Sende `selection$data()` og `selection$atlas()` inn til modul?

Reply:
Jeg lurte litt på hvordan det skal gjøres. Jeg tenkte først bare å sende inn data slik: data = selection. Så spesifiserer vi i modulen hvor den skal bruke atlas og data. Men det fungerer kanskje ikke?

Reply:
Ja, det vil kanskje fungere. Eventuelt sende de hver for seg, så er man mer fleksibel senere (hvis valg av atlas ikke lenger skal være en del av selection).

Reply:
Jeg får denne feilmeldingen. 
Error: no applicable method for 'st_transform' applied to an object of class "NULL

Det er vel koblet til denne tenker jeg. 
 map_to_plot <- sf::st_transform(healthatlas_data[[selection$atlas()]][["map"]], 32633)
Hvordan skrive om denne når jeg bruker den i modulen?

Reply:
Kartet som skal brukes ligger i `healthatlas_data$<atlas>$map`, så kanskje denne også burde sendes inn til modulen?
```R
tab_server("plots", atlas = ..., data = ..., map = ...)
```
Kanskje du ikke trenger å sende inn `atlas`?

Reply:
Så da erstatter du `healthatlas_data[[selection$atlas()]][["map"]]` med `map()` i modulen. `map` må være `reactive`, så mulig man må sende inn `map = reactive(healthatlas_data[[selection$atlas()]][["map"]])` eller lignende.

Reply:
Trenger vel bare map og data ja. 
Bare for å sjekke at jeg har forstått riktig. 
Modulen select returnerer to lister, data og atlas. 
I master hvor figurene ikke er i moduler brukes det healthatlas_data [[selection$atlas()]] for å velge riktig kart. 

Mange feilmeldinger i rødt her..
Det går litt sakte, prøver å forstå hva alle de ulike delene gjør. 
Det er problemer med å få sendt inn riktig kart i modulen.

Reply:
👯‍♂️ Endelig vart det riktig! Æ fikk opp kartet. Må bærre hold styr på alle parantesan i ulike varianta!

Reply:
Riktig e å ta litt hardt i..heilt riktig vart det ikkje. Fremdeles endel feil, men det kom hvertfall opp et kart!

Reply:
Modulen select returnerer en liste med to elementer, filtrert data og hvilket atlas som er valgt. Sistnevnte er kun en string med et navn (`"dagkir"`, `"barn"` eller lignende), som kan brukes videre.

`healthatlas_data`, som lages av funksjonen `get_data()`, inneholder alle data og alle kart. Det er en liste med lister... Øverste nivå med lister er atlas (`"dagkir"`, `"barn"` eller lignende). Fødsel ser f.eks slik ut:
```R
    "fodsel"   = list("data_no" = data::fodsel,
                      "data_en" = data::fodsel_en,
                      "year" = "2015\u20132017",
                      "title_no" = "F\u00f8dselshjelp",
                      "title_en" = "Obstetrics",
                      "map" = kart::fodsel)
```
så hvis man skal plotte kart avfødselsatlaset trenger man `kart::fodsel`, som allerede er lagt inn i `healthatlas_data$fodsel$map` (For å hente ut dette elementet ved å bruke tekststringen `"fodsel"` kan man heller skrive `healthatlas_data[["fodsel"]]$map`).